### PR TITLE
WIP: Add support for Gitee provider

### PR DESCRIPTION
**Pull request by lauvinson**
Please read [CONTRIBUTING.md](CONTRIBUTING.md) for details on our [code of conduct](CODE_OF_CONDUCT.md), and the process for submitting pull requests to us.

Fixes #

## Proposed Changes

- Add support for Gitee provider

### Checklist

- [x] Read the [**CONTRIBUTING**](CONTRIBUTING.md) document.
- [x] Read the [**CODE OF CONDUCT**](CODE_OF_CONDUCT.md) document.
- [x] Add tests to cover changes.
- [x] Ensure your code follows the code style of this project.
- [x] Ensure CI and all other PR checks are green OR
    - [x] Code compiles correctly.
    - [x] Created tests which fail without the change (if possible).
    - [x] All new and existing tests passed.
- [x] Add your changes to `Unreleased` section of [CHANGELOG](CHANGELOG.md).
- [x] Improve and update the [README](README.md) (if necessary).

### CODE REVIEW

File: providers/gitee.go [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+		Transport: &transport.BearerToken{
+			Token: token,
+		},
+	}
+
+	var changes []*scm.Change
+	var _ *scm.Response
+
+	if build.Before == "" || build.Before == scm.EmptyCommit {
+		changes, _, err = client.Git.ListChanges(newctx, repo.Slug, build.After, scm.ListOptions{})
```
Comment (jimsheldon) [inline providers/gitee.go:31]: Thanks for working on this!

Unfortunately git-scm does not currently have support for `ListChanges` (or `CompareChanges` below) https://github.com/drone/go-scm/blob/master/scm/driver/gitea/git.go#L76-L78

This is preventing us from supporting Gitea as well (see #81).

I think you will need to add that support to the git-scm module before this extension can support Gitee.

**Comment by jimsheldon** [comment]
Thanks for working on this! Looks like tests are failing, please let me know if you need help fixing them.

**Comment by lauvinson** [comment]
> Thanks for working on this! Looks like tests are failing, please let me know if you need help fixing them.

Excuse me
How do I pass the coverage test，Whether the testData folder is missing json files

**Comment by jimsheldon** [comment]
@lauvinson I usually just bring over the test json files I need from the go-scm library https://github.com/drone/go-scm/tree/master/scm/driver/gitee/testdata

**Comment by lauvinson** [comment]
@jimsheldon okay, I‘m done

**Comment by Satont** [comment]
Any updates?

**Comment by hikerspath** [comment]
I will take a look into this tomorrow / Thursday morning if Jim doesn't get to it first 😄

**Comment by lauvinson** [comment]
thankyou

Brian Burnett ***@***.***> 于2022年8月10日周三 03:35写道：

**Review by jimsheldon — APPROVED** [review]

**Comment by jimsheldon** [comment]
Finally found some time to review this, looks good! Thanks for the contribution @lauvinson!

**Pull request merged**